dart_io_mini_samples/example/files_directories_and_symlinks/getting_the_parent_directory.dart

```diff
@@ -7,12 +7,11 @@
 /// and Link.
 
 import 'dart:io';
+import 'dart:async'; // Import not needed but added here to explicitly assign type for clarity below.
 
-void main() {
+main() async {
   // List the contents of the system temp directory.
-  Directory.systemTemp.list(recursive: true, followLinks: false)
-    .listen((FileSystemEntity entity) {
-      // Print the path of the parent of each file, directory, and symlink.
-      print(entity.parent.path);
-    });
+  Stream<FileSystemEntity> entityList = Directory.systemTemp.list(recursive: true, followLinks: false);
+
+  await for (FileSystemEntity entity in entityList) print(entity.parent.path);
```

Comment:
I'm curious about whether the formatter would try to break this into multiple lines...